contracts/swapHandlers/UniswapV3SwapHandler.sol

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+pragma solidity ^0.8.0;
+
+import "./SwapHandlerBase.sol";
+import "../vendor/ISwapRouter.sol";
+
+/// @notice Swap handler executing trades on UniswapV3 through SwapRouter
+contract UniswapV3SwapHandler is SwapHandlerBase {
+    address immutable public uniswapRouter;
+
+    constructor(address uniswapRouter_) {
+        uniswapRouter = uniswapRouter_;
+    }
+
+    function executeSwap(SwapParams calldata params) override external {
+        setMaxAllowance(params.underlyingIn, uniswapRouter);
+
+        if (params.payload.length == 64) {
```

Comment:
Comment would be great here: What are these two paths for?

Reply:
:+1: